### PR TITLE
Migrates Reader remotes + models + tests into WordPressKit

### LINKED ISSUE #7323: Migrate ReaderPostServiceRemote to WordPressKit framework
Refer to: #7301

Migrate the remote service class along with any related data model classes and unit tests.

### LINKED ISSUE #7324: Migrate ReaderSiteServiceRemote to WordPressKit framework
Refer to: #7301

Migrate the remote service class along with any related data model classes and unit tests.

### LINKED ISSUE #7325: Migrate ReaderTopicServiceRemote to WordPressKit framework
Refer to: #7301

Migrate the remote service class along with any related data model classes and unit tests.

### PR DESCRIPTION
**Pull request by bummytime**
Fixes #7323 
Fixes #7324 
Fixes #7325 

**Models:**
* Migrates the `RemoteReaderPost.h/.m` model to the WPKit framework
* Migrates the `RemoteReaderSite.h/.m` model to the WPKit framework
* Migrates the `RemoteReaderSiteInfo.h/.m` model to the WPKit framework
* Migrates the `RemoteReaderTopic.h/.m` model to the WPKit framework
* Migrates the `RemoteSourcePostAttribution.h/.m` model to the WPKit framework
* Migrates the `RemoteReaderCrossPostMeta.swift` model to the WPKit framework

**Remotes:**
* Migrates the `ReaderPostServiceRemote.h/.m` and related tests into the WPKit framework
* Migrates the `ReaderSiteServiceRemote.h/.m` and related tests into the WPKit framework
* Migrates the `ReaderTopicServiceRemote.h/.m` and related tests into the WPKit framework

**WPShared:**
* Moves `BasePost`'s `summaryFromContent` class method to`NSString+Helpers.h/.m` in WPShared (including static const `PostDerivedSummaryLength`)
* Moves `testSummaryForContent` test to WPShared's `NSStringHelpersTests.m` (and deletes `BasePostTest.m` because that was the only unit test in that file)
* Migrates `PhotonImageURLHelper.h/.m` and related tests into the WPShared framework
* Migrates `DateUtils.h/.m` to the WPShared framework
* Migrates `RichContentFormatter.swift` and related tests into the WPShared framework (purposefully left 2 tests - `RichContentFormatterTests.swift` - in main WordPress app because of dependency on `UIApplication.shared.keyWindow`)
* Migrates `WPImageURLHelper.swift' to the WPShared framework

**Misc:**
* Updates related imports/refs in main WordPress project
* Migrates missed tests (NotificationSyncServiceRemoteTests.swift) from previous notes PR to WPKit

Ref #7301

**To test:**
1. Clean DerivedData.
2. Update CocoaPods repo with pod repo update.
3. Install the pods with pod install.
4. Run _all_ unit tests and make sure they are green.
5. Build & run the app:
a. Exercise the reader thoroughly. Try to test every aspect of its feature set.

@aerych given your supreme reader expertise, would you mind reviewing? _Greatly_ appreciated!

### CODE REVIEW

**Comment by aerych** [comment]
> @aerych given your supreme reader expertise, would you mind reviewing?

File: WordPressShared/WordPressShared/Core/Utility/NSString+Helpers.m [hunk 1]
```diff
@@ -112,6 +112,14 @@ + (NSString *)stripShortcodesFromString:(NSString *)string
                                       withTemplate:@""];
 }
 
++ (NSString *)summaryFromContent:(NSString *)string
```
Comment (aerych) [inline WordPressShared/WordPressShared/Core/Utility/NSString+Helpers.m]: I agree with factoring `summaryFromContent` out of `BasePost` but I'm not sure its generic enough to be a good addition to an `NSString` category/extension.  Its usecase is pretty specific.  The logic might have a better home in `ReaderPostServiceRemote createSummaryFromContent` and then we could just do away with the method.  As far as I can tell its not being called from anywhere else....

Comment (bummytime) [inline WordPressShared/WordPressShared/Core/Utility/NSString+Helpers.m]: @aerych I could get behind that. Let me move it into `createSummaryFromContent`.

Comment (bummytime) [inline WordPressShared/WordPressShared/Core/Utility/NSString+Helpers.m]: @aerych, Doh! So it looks like there is one other spot this method is called:

https://github.com/wordpress-mobile/WordPress-iOS/pull/7572/files#diff-fab690f488774f7082db0c4344d1e60b

Thoughts?

Comment (aerych) [inline WordPressShared/WordPressShared/Core/Utility/NSString+Helpers.m]: doH indeed. 
I still think its too specific for an NSString category but I don't have a better idea atm. :/   Let's go with the change as is and if we think of something better later we can revisit.  Cool?

Comment (bummytime) [inline WordPressShared/WordPressShared/Core/Utility/NSString+Helpers.m]: Wfm! Thank you!

**Pull request merged**